test/tests/functional/pbs_multi_sched.py

```diff
@@ -0,0 +1,601 @@
+                            a, id="sc2")
+        self.sched_configure('sc2', '15051', '/var/spool/pbs/sched_dir')
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 1,
+                             'sched_port': 15051}, id="sc2")
+        a = {'partition': 'P3',
+             'sched_host': self.server.hostname,
+             'sched_port': '15052'}
+        self.server.manager(MGR_CMD_CREATE, SCHED,
+                            a, id="sc3")
+        self.sched_configure('sc3', '15052')
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 1,
+                             'sched_port': 15052}, id="sc3")
+
+        a = {'queue_type': 'execution',
+             'started': 't',
+             'enabled': 't'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='wq1')
+        print self.server.queues
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='wq2')
+        print self.server.queues
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='wq3')
+        print self.server.queues
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='wq4')
+        p1 = {'partition': 'P1'}
+        self.server.manager(MGR_CMD_SET, QUEUE, p1, id='wq1')
+        p2 = {'partition': 'P2'}
+        self.server.manager(MGR_CMD_SET, QUEUE, p2, id='wq2')
+        p3 = {'partition': 'P3'}
+        self.server.manager(MGR_CMD_SET, QUEUE, p3, id='wq3')
+        p4 = {'partition': 'P4'}
+        self.server.manager(MGR_CMD_SET, QUEUE, p4, id='wq4')
+        a = {'resources_available.ncpus': 2}
+        self.server.create_vnodes('vnode', a, 5, self.mom)
+        self.server.manager(MGR_CMD_SET, NODE, p1, id='vnode[0]', expect=True)
+        self.server.manager(MGR_CMD_SET, NODE, p2, id='vnode[1]', expect=True)
+        self.server.manager(MGR_CMD_SET, NODE, p3, id='vnode[2]', expect=True)
+        self.server.manager(MGR_CMD_SET, NODE, p4, id='vnode[3]', expect=True)
+
+    def sched_configure(self, sched_name, sched_port=None, sched_home=None):
+        pbs_home = self.server.pbs_conf['PBS_HOME']
+        if sched_home is None:
+            sched_home = pbs_home
+        sched_priv_dir = 'sched_priv_' + sched_name
+        sched_logs_dir = 'sched_logs_' + sched_name
+        if not os.path.exists(os.path.join(sched_home, sched_priv_dir)):
+            self.du.run_copy(self.server.hostname,
+                             os.path.join(pbs_home, 'sched_priv'),
+                             os.path.join(sched_home, sched_priv_dir),
+                             recursive=True)
+        if not os.path.exists(os.path.join(sched_home, sched_logs_dir)):
+            self.du.run_copy(self.server.hostname,
+                             os.path.join(pbs_home, 'sched_logs'),
+                             os.path.join(sched_home, sched_logs_dir),
+                             recursive=True)
+        self.server.schedulers[sched_name].start(sched_port, sched_home)
+
```

Comment:
function "sched_configure" you are writing two times in functional test cases and interface test cases.
Suggest to put in the common place "PBSTestSuite" and may be make it more generic.